public/components/trace_analytics/components/traces/span_detail_table.tsx

```diff
@@ -253,67 +295,123 @@ export function SpanDetailTable(props: SpanDetailTableProps) {
 }
 
 export function SpanDetailTableHierarchy(props: SpanDetailTableProps) {
-  const { http, hiddenColumns, mode, DSL, dataSourceMDSId, availableWidth, openFlyout } = props;
+  const { hiddenColumns, mode, availableWidth, openFlyout } = props;
   const [items, setItems] = useState<Span[]>([]);
   const [_total, setTotal] = useState(0);
   const [expandedRows, setExpandedRows] = useState(new Set<string>());
   const [isSpansTableDataLoading, setIsSpansTableDataLoading] = useState(false);
 
   useEffect(() => {
-    setIsSpansTableDataLoading(true);
-    const spanSearchParams = {
-      from: 0,
-      size: 10000,
-      sortingColumns: [],
-    };
-    handleSpansRequest(
-      http,
-      (data) => {
-        const hierarchy = buildHierarchy(data);
-        setItems(hierarchy);
-      },
-      setTotal,
-      spanSearchParams,
-      DSL,
-      mode,
-      dataSourceMDSId
-    ).finally(() => setIsSpansTableDataLoading(false));
-  }, [DSL, http, mode, dataSourceMDSId]);
+    if (!props.payloadData) return;
+    try {
+      const parsed = JSON.parse(props.payloadData);
+      const hitsArray =
+        parsed.hits && Array.isArray(parsed.hits.hits)
+          ? parsed.hits.hits
+          : Array.isArray(parsed)
+          ? parsed
+          : [];
+
+      let spans = hitsArray.map((hit: any) => hit._source);
+
+      if (props.filters.length > 0) {
+        spans = spans.filter((span: any) => {
+          return props.filters.every(
+            ({ field, value }: { field: string; value: any }) => span[field] === value
```

Comment:
will there be negate filters like `field a is not b`?

Reply:
Not currently supported the flyout only allows filters to be added as such:
<img width="456" alt="Screenshot 2025-02-10 at 4 19 45 PM" src="https://github.com/user-attachments/assets/011d4d60-e469-491b-9849-8e87730ee608" />
Where the user can only clear. Intended use is for investigating based on the flyout 
<img width="385" alt="Screenshot 2025-02-10 at 4 20 37 PM" src="https://github.com/user-attachments/assets/06272d61-a117-48cc-b3f9-9bff6ffd9c7f" />